xray/segment.go

```diff
@@ -283,6 +301,11 @@ func (seg *Segment) CloseAndStream(err error) {
 	seg.Lock()
 	defer seg.Unlock()
 
+	// If segment is dummy we return
+	if seg.Dummy {
```

Comment:
At creating segment we dealt with the segment children membership no matter dummy or not. If we return early here, would skipping the following code cause mem leak? i.e. it will always be "in progress" and the parent always keep the reference of it

Reply:
it should be semantic